### PR TITLE
Add GraphNode 'selected' and 'unselected' signals, simplify GraphEdit

### PR DESCRIPTION
**Pull request by maximkulkin**
Backport of #64840 from master to 3.x

### CODE REVIEW

File: scene/gui/graph_edit.cpp [hunk 1]
```diff
@@ -402,6 +415,8 @@ void GraphEdit::add_child_notify(Node *p_child) {
 	if (gn) {
 		gn->set_scale(Vector2(zoom, zoom));
 		gn->connect("offset_changed", this, "_graph_node_moved", varray(gn));
+		gn->connect("selected", this, "_graph_node_selected", varray(gn));
```
Comment (akien-mga) [inline scene/gui/graph_edit.cpp:418]: I don't see method bindings in `_bind_methods` which is necessary to connect callbacks by string, so this wouldn't work.

Comment (maximkulkin) [reply scene/gui/graph_edit.cpp:418]: Fixed

**Review by YuriSizov — APPROVED** [review]
I think it's fine now.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**